src/plugins/intel_gpu/src/plugin/ops/non_max_suppression.cpp

```diff
@@ -52,7 +52,8 @@ static void CreateNonMaxSuppressionIEInternalOp(ProgramBuilder& p, const std::sh
 
     auto boxesShape = op->get_input_partial_shape(0);
     size_t num_outputs = op->get_output_size();
-    if (p.use_new_shape_infer()) {
+    if (p.use_new_shape_infer() && op->get_output_partial_shape(0).is_dynamic()) {
```

Comment:
Now the model is not going to use InternalOp when it is dynamic model. Do we still need this pass?

Reply:
@yeonbok  You're right. The path isn't need for current transformation pipeline. But I'd like to keep full coverage in NMSIEInternal op creation.